ballerina/http_client_payload_builder.bal

```diff
@@ -161,15 +162,15 @@ isolated function jsonPayloadBuilder(Response response, TargetType targetType) r
 isolated function nonNilablejsonPayloadBuilder(Response response, typedesc<anydata> targetType)
         returns anydata|ClientError {
     json payload = check response.getJsonPayload();
-    var result = payload.fromJsonWithType(targetType);
+    var result = jsondata:parseAsType(payload, {enableConstraintValidation: false}, targetType);
```

Comment:
By default, `parseAsType` is enabled projection support. Since the `fromJsonWithType` does not have the projection support, we need to disabled it by default.
```suggestion
    var result = jsondata:parseAsType(payload, {enableConstraintValidation: false, allowDataProjection: false}, targetType);
```